### ISSUE TITLE
sonar-redmine can work with  Redmine 3.0.3

### ISSUE THREAD

**Issue by akeelahmedqureshi** [SonarCommunity/sonar-redmine#14]
Hello,

Could you please let me know sonar-redmine can work with  Redmine 3.0.3(Rails 4) ?

Thanks
Akeel

**Comment by CSchulz** [comment]
I would not guarantee it. The plugin is build against the Redmine REST API 1.19 and it is tested with Redmien 2.2.0 and 2.3.0.

**Comment by jpmolekunnel** [comment]
Hello,

Error : "Missing parameter: resource"

While integrating Sonar with Redmine, At Project level, go to Administration -> Redmine Configuration Page is not loading, Instead of that am getting the above error message on browser.

I am using following versions,
SonarQube 5.2 and Redmin 3.0.4

Please help me on this,

Regards,
Jojan Paul

**Comment by ogg1e** [comment]
I'm seeing the same error:
Error : "Missing parameter: resource"

but I'm on redmine 2.5.2 with sonarqube 5.2

**Comment by omji** [comment]
Same issue with "Missing parameter: resource"
Redmine 2.6 & sonarqube 5.2

**Comment by omji** [comment]
For those who have the same issue, I have found a workaround until this is fixed. In the top url, you need to replace the GET paramater 'id' by 'resource'. 

So insterad of 

http://{sonar_url}/redmine_configuration?id={project}

Replace 'id' with 'resource'

http://{sonar_url}/redmine_configuration?resource={project}

**Comment by sergpr84** [comment]
Thanks omji. It helps.
Maybe you know. I try to connect redmine with sonarqube, but when I click on "Link to redmine" I have an error: A problem occurred with the remote server:Project cannot be blank.
If you can help, please help.
More info about this here: http://stackoverflow.com/questions/34897471/sonarqube-redmine-plugin-connection-error

**Comment by omji** [comment]
Hi,

I have answered your question on stackoverflow. You basically need to setup the projet key and make sure you hit 'save settings'.

BUT! Once you connect succesfully, you still wont be fully done unfortunately :-( The plugin is outdated so the link it generates in the redmine ticket wont work.

In order to fix it, I had to cheat with a dirty hack... If you want to do the same, you will need to get the code from github, then modify this file: https://github.com/SonarQubeCommunity/sonar-redmine/blob/master/src/main/java/org/sonar/plugins/redmine/reviews/RedmineIssueFactory.java

Line 71, change sb.append("/issue/show/"); to sb.append("/issues/search#issues=");

And then comes the tricky part, you will need to compile using maven! And replace the plugin file in youyr sonar install with the .jar you generated, in my case: /opt/sonar/extensions/plugins/sonar-redmine-plugin-0.2.jar

If you want I can give you my fixed .jar, although I am not a Java / Maven dev. It works fine for me, but not sure I did it the best way.

Once again its not ideal, but hope it helps :-(

**Comment by sergpr84** [comment]
Thank you very much for your answer. I found solution: http://stackoverflow.com/questions/34897471/sonarqube-redmine-plugin-connection-error Now I have a new problem. On my new created issue, I have a bad link to sonar, something like this: 
http://sonarqube_url:9000/issue/show/81e1d47f-d0b2-4276-9244-9470408eb1cf The page you were looking for doesn't exist. You may have mistyped the address or the page may have moved. I will try to found solution for it too :) Also I can say that this plugin works with redmine 3.1.2.stable and sonarqube 5.3 :) 
16 февраля 2016, 12:02:48, от "Guillaume Pousseo" < notifications@github.com >: 

Hi, I have answered your question on stackoverflow. You basically need to setup the projet key and make sure you hit 'save settings'. BUT! Once you connect succesfully, you still wont be fully done unfortunately :-( The plugin is outdated so the link it generates in the redmine ticket wont work. In order to fix it, I had to cheat with a dirty hack... If you want to do the same, you will need to get the code from github, then modify this file: https://github.com/SonarQubeCommunity/sonar-redmine/blob/master/src/main/java/org/sonar/plugins/redmine/reviews/RedmineIssueFactory.java Line 71, change sb.append("/issue/show/"); to sb.append("/issues/search#issues="); And then comes the tricky part, you will need to compile using maven! And replace the plugin file in youyr sonar install with the .jar you generated, in my case: /opt/sonar/extensions/plugins/sonar-redmine-plugin-0.2.jar If you want I can give you my fixed .jar, although I am not a Java / Maven dev. It works fine for me, but not sure I did it the best way. Once again its not ideal, but hope it helps :-( —

**Comment by omji** [comment]
I have answered this last 404 issue already, I knew you would face it. Please read my previous post again :-)

**Comment by sergpr84** [comment]
Yes, you are wright! Maybe you khow how to resolve a problem with link in redmine issue? 
Here is the screens: 
1. http://joxi.ru/EA4kDl6sDMQVK2 2. http://joxi.ru/L210DkRf6abKKm 

16 февраля 2016, 12:22:44, от "Guillaume Pousseo" < notifications@github.com >: 

I have answered this last issue already, I knew you would face it. Please read my previous post again :-) —

**Comment by omji** [comment]
Yes, I have explained in details how to solve this: which code line to change, how to compile and re-install plugin. I even proposed my .jar file. Please read my comment again, its all there.

**Comment by sergpr84** [comment]
I can`t find a .jar file. Please give me your fixed .jar file.  Thanks :) 

16 февраля 2016, 13:04:42, от "Guillaume Pousseo" < notifications@github.com >: 

Yes, I have explained in details how to solve this: which code line to change, how to compile and re-install plugin. I even proposed my .jar file. Please read my comment again, its all there. —

**Comment by omji** [comment]
Hi,

I didnt post it, as I said I wasn't sure you wanted it. Once again, I am not a java / maven specialist. File is in the attached zip.

[sonar-redmine-issue-url-fix.zip](https://github.com/SonarQubeCommunity/sonar-redmine/files/133962/sonar-redmine-issue-url-fix.zip)

**Comment by star9264** [comment]
Hello
I have the same error
Error : "Missing parameter: resource"

but I really want to modify the plugin source or sonarqube source. How can I do?

**Comment by ghost** [comment]
add "s" to string:
old string:   message.append("/issue/");  should be: message.append("/issues/");
in file: sonar-redmine-plugin\src\main\java\org\sonar\plugins\redmine\reviewsRedmineLinkFunction.java
Then, build .jar file:
1. delete folder "target" in plugin folder
2. cd path_to_plugin\sonar-redmine-plugin
3. mvn package -Dmaven.test.skip=true
